sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/utils-service/_tests_/unit/application-event-code-utils.service.getApplicationEventCode.spec.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  ApplicationStatus,
+  COEStatus,
+  DisbursementSchedule,
+  DisbursementScheduleStatus,
+  RestrictionActionType,
+} from "@sims/sims-db";
+import { ApplicationEventCode } from "../../../models/ier12-integration.model";
+import {
+  ApplicationEventCodeDuringAssessmentUtilsService,
+  ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService,
+  ApplicationEventCodeUtilsService,
+} from "../..";
+import { DATE_ONLY_ISO_FORMAT, formatDate } from "@sims/utilities";
+import { createMock } from "@golevelup/ts-jest";
+
+describe("ApplicationEventCodeUtilsService-getApplicationEventCode", () => {
+  let applicationEventCodeUtilsService: ApplicationEventCodeUtilsService;
+  let applicationEventCodeDuringAssessmentUtilsService: ApplicationEventCodeDuringAssessmentUtilsService;
+  let applicationEventCodeDuringEnrolmentAndCompletedUtilsService: ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService;
+  let payload: Pick<
+    DisbursementSchedule,
+    "id" | "coeStatus" | "disbursementDate" | "disbursementScheduleStatus"
+  >;
+  let applicationNumber: string;
+  let activeRestrictionsActionTypes: RestrictionActionType[][];
+
+  beforeAll(() => {
+    applicationEventCodeDuringAssessmentUtilsService =
+      createMock<ApplicationEventCodeDuringAssessmentUtilsService>();
+    applicationEventCodeDuringEnrolmentAndCompletedUtilsService =
+      createMock<ApplicationEventCodeDuringEnrolmentAndCompletedUtilsService>();
+    applicationEventCodeUtilsService = new ApplicationEventCodeUtilsService(
+      applicationEventCodeDuringAssessmentUtilsService,
+      applicationEventCodeDuringEnrolmentAndCompletedUtilsService,
+    );
+    // Arrange.
+    payload = {
+      id: 9999,
+      coeStatus: COEStatus.completed,
+      disbursementDate: formatDate(new Date(), DATE_ONLY_ISO_FORMAT),
+      disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+    };
+    applicationNumber = "999999999";
+    activeRestrictionsActionTypes = [];
+  });
+
+  it(`Should return ${ApplicationEventCode.DISC} when the application status is ${ApplicationStatus.Cancelled}.`, async () => {
+    // Act
+    const applicationEventCode =
+      await applicationEventCodeUtilsService.getApplicationEventCode(
+        applicationNumber,
+        ApplicationStatus.Cancelled,
+        payload,
+      );
+    // Assert
+    expect(applicationEventCode).toBe(ApplicationEventCode.DISC);
+  });
+
+  it(`Should call applicationEventCodeDuringAssessment when the application status is ${ApplicationStatus.Assessment}.`, async () => {
+    // Act
+    await applicationEventCodeUtilsService.getApplicationEventCode(
+      applicationNumber,
+      ApplicationStatus.Assessment,
+      payload,
+    );
+    // Assert
+    expect(
```

Comment:
To clarify, isn't our intention here to assert the final event code obtained? I know we don't have to cover all the scenrios which happens inside this method as they are covered in the unit tests for child methods.